pvs/V1028 screen.c:6083

ngortheone opened the pull request:
Fixing instances of V1028 from PVS report labeled as level high, one at a time

As per @justinmk request here https://github.com/neovim/neovim/pull/10478

ngortheone committed:
pvs/V1028 screen.c:6083

bfredl commented on src/nvim/screen.c:
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
   ScreenGrid new = *grid;
-
-  size_t ncells = (size_t)((rows+1) * columns);
+  /* Since amount of cells could not be negative we expect both
```
use `//` comments.

bfredl commented on src/nvim/screen.c:
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
+  /* Since amount of cells could not be negative we expect both
+   * rows and columns to be positive */
+  assert(rows >= 0 && columns >= 0);
+  size_t ncells = ((size_t)rows + 1) * columns;
```
the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

ngortheone replied:
done

justinmk replied:
> the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

that explanation should be pasted into the commit message.

ngortheone committed:
code review

@bfedl:
the + 1 can be removed. it was used for current_screenline, which
now is dedicated linebuf_char buffer.

justinmk commented on src/nvim/screen.c:
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
+  // Since amount of cells could not be negative we expect both
+  // rows and columns to be positive
+  assert(rows >= 0 && columns >= 0);
+  size_t ncells = (size_t)rows  * columns;
```
too many spaces.

ngortheone replied:
fixed

justinmk commented:
> the `+ 1` can be removed. it was used for `current_screenline`, which now is dedicated `linebuf_char` buffer.

that explanation should be pasted into the commit message.

Commit:
Merge branch 'master' into pvs/V1028_screen_c_6083

Commit:
Merge branch 'master' into pvs/V1028_screen_c_6083

ngortheone commented:
> that explanation should be pasted into the commit message.

@justinmk done

justinmk commented on src/nvim/screen.c:
```diff
@@ -6079,8 +6079,10 @@ void grid_alloc(ScreenGrid *grid, int rows, int columns, bool copy, bool valid)
-
-  size_t ncells = (size_t)((rows+1) * columns);
+  // Since amount of cells could not be negative we expect both
+  // rows and columns to be positive
```
the comment is redundant with the  `assert()`

ngortheone replied:
removed

Commit:
Merge branch 'pvs/V1028_screen_c_6083' of github.com:ngortheone/neovim into pvs/V1028_screen_c_6083

justinmk commented:
Thanks. For PVS commit messages I usually include some form of the PVS warning or advice. For V1028 that is "cast operands, not the result".

ngortheone commented:
ack. will add to other commit messages.
